### PR TITLE
dvc.yaml: document `persist` field, et al.

### LINKED ISSUE #2027: guide: doc persist field in dvc.yaml, maybe improve explanation in run --outs-persist
> Context: https://discuss.dvc.org/t/link-types-and-run-options/590/4

Per iterative/dvc/issues/1214 (iterative/dvc/pull/1759)

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Closes #2027

### CODE REVIEW

File: content/docs/user-guide/dvc-files-and-directories.md [hunk 1]
```diff
@@ -75,10 +75,13 @@ An _output entry_ (`outs`) consists of these fields:
 - `nfiles`: If a directory, number of files inside.
 - `cache`: Whether or not this file or directory is <abbr>cached</abbr> (`true`
   by default, if not present). See the `--no-commit` option of `dvc add`.
+- `persist`: Whether the output file/dir should remain in place while
+  `dvc repro` runs. By default outputs are deleted when `dvc repro` starts (if
+  this value is not present).
```
Comment (jorgeorpinel) [inline content/docs/user-guide/dvc-files-and-directories.md:80]: All the other changes to this doc are unrelated, sorry. They're just copy edits along-the-way.

File: content/docs/command-reference/import.md [hunk 2]
```diff
@@ -152,7 +152,6 @@ outs:
   - md5: a304afb96060aad90176268345e10355
     path: data.xml
     cache: true
-    metric: false
```
Comment (jorgeorpinel) [inline content/docs/command-reference/import.md:155]: This one is also an unrelated fix.

Comment (jorgeorpinel) [reply content/docs/command-reference/import.md:155]: (same in content/docs/user-guide/external-dependencies.md)

File: content/docs/command-reference/run.md [hunk 3]
```diff
@@ -185,7 +186,8 @@ $ dvc run -n my_stage './my_script.sh $MYENVVAR'
   reason.
 
 - `--outs-persist <path>` - declare output file or directory that will not be
-  removed upon `dvc repro`.
+  removed when `dvc repro` starts (but it will still be overwritten when it
```
Comment (shcheklein) [inline content/docs/command-reference/run.md:189]: this is not correct ... not necessarily overwritten. It can stay the same, it can be appended in case of a dir, etc

File: content/docs/user-guide/dvc-files-and-directories.md (hunk 1 above)
Comment (shcheklein) [reply content/docs/user-guide/dvc-files-and-directories.md:80]: please, please, let's not do this :)

**Review by shcheklein — APPROVED** [review]

File: content/docs/user-guide/dvc-files-and-directories.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/dvc-files-and-directories.md:80]: OK

File: content/docs/command-reference/run.md (hunk 3 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/run.md:189]: True. Changing to "but it can still be modified, overwritten, or even deleted by the stage command(s)" in my Git stash for now.

Comment (shcheklein) [reply content/docs/command-reference/run.md:189]: I think it's fine to omit this completely. That's expected by default to my mind.

Comment (jorgeorpinel) [reply content/docs/command-reference/run.md:189]: I personally had to try this to see what it did. It wasn't obvious what to expect. It's not that clear by just saying "will not be removed". A lot of users are surprised to learn that DVC deletes outputs before repro in the first place.

Comment (shcheklein) [reply content/docs/command-reference/run.md:189]: kk, up to you on this one!
